examples/baselines/act/train_rgbd.py

```diff
@@ -0,0 +1,607 @@
+    lr_backbone: float = 1e-5
+    masks: bool = False
+    dilation: bool = False
+    include_depth: bool = True
+
+    # Transformer
+    enc_layers: int = 4
+    dec_layers: int = 7
+    dim_feedforward: int = 1600
+    hidden_dim: int = 512
+    dropout: float = 0.1
+    nheads: int = 8
+    num_queries: int = 30
+    pre_norm: bool = False
+
+    # Environment/experiment specific arguments
+    max_episode_steps: Optional[int] = None
+    """Change the environments' max_episode_steps to this value. Sometimes necessary if the demonstrations being imitated are too short. Typically the default
+    max episode steps of environments in ManiSkill are tuned lower so reinforcement learning agents can learn faster."""
+    log_freq: int = 1000
+    """the frequency of logging the training metrics"""
+    eval_freq: int = 5000
+    """the frequency of evaluating the agent on the evaluation environments"""
+    save_freq: Optional[int] = None
+    """the frequency of saving the model checkpoints. By default this is None and will only save checkpoints based on the best evaluation metrics."""
+    num_eval_episodes: int = 100
+    """the number of episodes to evaluate the agent on"""
+    num_eval_envs: int = 10
+    """the number of parallel environments to evaluate the agent on"""
+    sim_backend: str = "cpu"
+    """the simulation backend to use for evaluation environments. can be "cpu" or "gpu"""
+    num_dataload_workers: int = 0
+    """the number of workers to use for loading the training data in the torch dataloader"""
+    control_mode: str = 'pd_joint_delta_pos'
+    """the control mode to use for the evaluation environments. Must match the control mode of the demonstration dataset."""
+
+    # additional tags/configs for logging purposes to wandb and shared comparisons with other algorithms
+    demo_type: Optional[str] = None
+
+
+class FlattenRGBDObservationWrapper(gym.ObservationWrapper):
+    """
+    Flattens the rgbd mode observations into a dictionary with two keys, "rgbd" and "state"
+
+    Args:
+        rgb (bool): Whether to include rgb images in the observation
+        depth (bool): Whether to include depth images in the observation
+        state (bool): Whether to include state data in the observation
+
+    Note that the returned observations will have a "rgbd" or "rgb" or "depth" key depending on the rgb/depth bool flags.
+    """
+
+    def __init__(self, env, rgb=True, depth=True, state=True) -> None:
+        self.base_env: BaseEnv = env.unwrapped
+        super().__init__(env)
+        self.include_rgb = rgb
+        self.include_depth = depth
+        self.include_state = state
+        self.transforms = T.Compose(
+            [
+                T.Resize((224, 224), antialias=True),
+            ]
+        )  # resize the input image to be at least 224x224
+        new_obs = self.observation(self.base_env._init_raw_obs)
+        self.base_env.update_obs_space(new_obs)
+
+    def observation(self, observation: Dict):
```

Comment:
is there a reason why the maniskill provided flatten rgbd wrapper cant be used? save some code here.

Reply:
1. Visual Data Shape: 
    The visual data (rgb or rgbd) is of shape [batch, num_cams, C, H, W] instead of [batch, H, W, num_cams*C] as it makes it easier to  use as input to CNN backbone. The CNN backbone accepts input of size [batch, C, H, W] so we query the backbone for each camera image and concatenate them afterwards.

2. Visual Data Size: 
    The pre-trained models from torchvision.models expect input image to be at least 224x224, and the default visual data size is 128x128. I learned recently that you can change this with sensor_config param in gym.make but if this only changes the rgbd returned by the env and not the rgbd from the replayed trajectory dataset, we would still need to resize the rgbd from the replayed trajectory dataset, and there would be a discrepancy between the two rgbd. I'm not sure how much this discrepancy will affect the performance as I haven't tested them, but resizing it in the wrapper makes it consistent.

Reply:
Ok no worries then.